repository/src/lib.rs

```diff
@@ -439,10 +439,67 @@ impl<T: RawRepository> DistributedRepository<T> {
     /// and update the corresponding `a-#` branch to it
     pub async fn approve(
         &mut self,
-        _agenda_commit_hash: &CommitHash,
-        _proof: Vec<(PublicKey, TypedSignature<Agenda>)>,
+        agenda_commit_hash: &CommitHash,
+        proof: Vec<(PublicKey, TypedSignature<Agenda>)>,
     ) -> Result<CommitHash, Error> {
-        unimplemented!()
+        let last_header = self.get_last_finalized_block_header().await?;
+        let last_header_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+        // Check if the agenda is rebased on top of the `finalized` branch.
+        if self
+            .raw
+            .find_merge_base(last_header_commit, *agenda_commit_hash)
+            .await?
+            != last_header_commit
+        {
+            return Err(anyhow!(
+                "agenda:{} should be rebased on {}",
+                agenda_commit_hash,
+                FINALIZED_BRANCH_NAME
+            ));
+        }
+
+        // Verify agenda with agenda proof
+        let agenda_commit = self.read_commit(*agenda_commit_hash).await?;
+        let agenda = match agenda_commit {
+            Commit::Agenda(agenda) => agenda,
+            _ => panic!("given commit hash is not agenda commit hash"),
+        };
+        for (_, typedsignature) in proof.clone() {
+            typedsignature.verify(&agenda)?;
+        }
+
+        // Delete past a-#(agenda) branch and create new a-#(agendaproof) branch
+        self.raw
+            .delete_branch(format!(
+                "a-{:?}",
+                agenda_commit_hash
+                    .to_string()
+                    .truncate(TAG_NAME_HASH_DIGITS)
```

Comment:
```suggestion
                agenda_commit
                    .to_hash256()
                    .to_string()
                    .truncate(BRANCH_NAME_HASH_DIGITS)
```
wrong implementation
also add BRANCH_NAME_HASH_DIGITS please

Reply:
fixed.